### ISSUE TITLE
node/coordinator communication & authentication

### ISSUE THREAD

**Issue by profi248** [profi248/pluto#2]
response messages from BC will contain a success boolean, and in case of an error will have a predefined code (optionally a string)

session establishment/authentication:

```
node connects with client id (sid): base32(rand(128))

[coordinator/auth] node -> coordinator
{ pk: node_pubkey }

[node/<sid>/auth] coordinator -> node
{ challenge: rand(256) }

[coordinator/auth] node -> coordinator
{ response: sign_sk(challenge) }

[node/<sid>/auth] coordinator -> node
{ session_token: rand(256) }
```
(in case sid already exists, coordinator won't respond, node will resend a new request after a timeout)

**Comment by profi248** [comment]
well, the sender client ID is not sent along with the message in MQTT, so this is stupid and doesn't work in this state

**Comment by profi248** [comment]
in the end, it's implemented in a completely different way

**Comment by profi248** [comment]
authentication implemented in c88c7f717
